File: app/src/main/java/com/example/ikastyle/Dao/CustomizationMainDao.java
package com.example.ikastyle.Dao;

import androidx.room.Dao;
import androidx.room.Query;

import com.example.ikastyle.DatabaseView.CustomizationMain;

import java.util.List;

/*
 * WEAPON_MAINビューのDAO
 */
@Dao
public interface CustomizationMainDao {
    @Query("SELECT * FROM CUSTOMIZATION_MAIN_NAME WHERE language_code = :languageCode ORDER BY category_id, main_name, weapon_id")
    List<CustomizationMain> getWeaponMainList(int languageCode);
}

File: app/src/main/java/com/splatool/ikastyle/model/data/dao/CustomizationMainDao.kt
package com.splatool.ikastyle.model.data.dao

import com.splatool.ikastyle.model.data.databaseView.CustomizationMain
import androidx.room.*

/*
 * WEAPON_MAINビューのDAO
 */
@Dao
interface CustomizationMainDao {
    @Query("SELECT * FROM CUSTOMIZATION_MAIN_NAME WHERE language_code = :languageCode ORDER BY category_id, main_name, weapon_id")
    fun getWeaponMainList(languageCode: Int): List<CustomizationMain>

    @Query("SELECT * FROM CUSTOMIZATION_MAIN_NAME WHERE language_code = :languageCode AND category_id = :categoryId ORDER BY category_id, main_name, weapon_id")
    fun getWeaponMainListByCategory(languageCode: Int, categoryId : Int): List<CustomizationMain>
}